c/tests/test_tables.c

```diff
@@ -4760,47 +4760,176 @@ test_link_ancestors_multiple_to_single_tree(void)
 }
 
 static void
-test_ibd_finder(void)
+verify_ibd_segment_list(tsk_segment_list_t *list, tsk_size_t num_nodes)
+{
+    tsk_segment_t *seg;
+    double total_span = 0;
+    tsk_size_t num_segments = 0;
+    /* double last_right = 0; */
+
+    for (seg = list->head; seg != NULL; seg = seg->next) {
+        CU_ASSERT_FATAL(seg->left < seg->right);
+        CU_ASSERT_FATAL(seg->node >= 0);
+        CU_ASSERT_FATAL(seg->node < (tsk_id_t) num_nodes);
+        total_span += seg->right - seg->left;
+        num_segments++;
+
+        /* TODO the segments are not necessarily in order - issue #1682 */
```

Comment:
I think this is expected, though I know it's annoying. I'll comment more in the issue about it.